AsyncDisplayKit/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -0,0 +1,153 @@
+  self.accessibilityLabel = node.accessibilityLabel;
+  self.accessibilityHint = node.accessibilityHint;
+  self.accessibilityValue = node.accessibilityValue;
+  self.accessibilityTraits = node.accessibilityTraits;
+}
+
+- (ASDisplayNode *)asyncdisplaykit_node
+{
+  return objc_getAssociatedObject(self, ASDisplayNodeAssociatedNodeKey);
+}
+
+@end
+
+
+#pragma mark - _ASDisplayView
+
+@interface _ASDisplayView () {
+  NSMutableArray *_accessibleElements;
+}
+
+@end
+
+@implementation _ASDisplayView (UIAccessibilityContainer)
+
+#pragma mark - UIAccessibility
+
+- (NSArray *)accessibleElements
+{
+  ASDisplayNode *selfNode = self.asyncdisplaykit_node;
+  if (selfNode == nil) {
+    return nil;
+  }
+  
+  _accessibleElements = [[NSMutableArray alloc] init];
+  
+  // Handle rasterize case
+  if (selfNode.shouldRasterizeDescendants) {
+    // In this case we have to go through the whole subnodes tree in BFS fashion and create all
+    // accessibility elements ourselves as the view hierarchy is flattened
+    
+    // Queue used to keep track of subnodes while traversing this layout in a BFS fashion.
+    std::queue<ASDisplayNode *> queue;
+    queue.push(selfNode);
+    
+    while (!queue.empty()) {
+      ASDisplayNode *node = queue.front();
+      queue.pop();
+      
+      // Check if we have to add the node to the accessiblity nodes as it's an accessiblity element
+      if (node != selfNode && node.isAccessibilityElement) {
+        UIAccessibilityElement *accessibilityElement = [[UIAccessibilityElement alloc] initWithAccessibilityContainer:self];
+        accessibilityElement.asyncdisplaykit_node = node;
+        [_accessibleElements addObject:accessibilityElement];
+      }
+
+      // Add all subnodes to process in next step
+      for (int i = 0; i < node.subnodes.count; i++)
+        queue.push(node.subnodes[i]);
+    }
+    return _accessibleElements;
+  }
+  
+  // Handle not rasterize case
+  // Create UI accessiblity elements for each subnode that represent an elment within the accessibility container
+  for (ASDisplayNode *subnode in selfNode.subnodes) {
+      // Check if this subnode is a UIAccessibilityContainer
+    if (!subnode.isAccessibilityElement && [subnode accessibilityElementCount] > 0) {
+      // We are good and the view is an UIAccessibilityContainer so add it
+      [_accessibleElements addObject:subnode.view];
+    } else if (subnode.isAccessibilityElement) {
+      // Create a accessiblity element from the subnode
+      UIAccessibilityElement *accessibilityElement = [[UIAccessibilityElement alloc] initWithAccessibilityContainer:self];
+      accessibilityElement.asyncdisplaykit_node = subnode;
+      [_accessibleElements addObject:accessibilityElement];
+    }
+  }
+  
+  return _accessibleElements;
+}
+
+- (NSInteger)accessibilityElementCount
+{
+  return [self accessibleElements].count;
+}
+
+- (id)accessibilityElementAtIndex:(NSInteger)index
+{
+  if (_accessibleElements == nil) {
```

Comment:
In this case, maybe we should actually trigger the creation of accessibleElements?  I am not sure what to expect of the access pattern for this method, but especially because I think it is no longer used in iOS 9 / thus we will get very little testing, it might be safer to trigger creating.  I honestly don't know though / am not familiar with when the system must call this method and what it would do if nil.